src/main/perl/Filesystem.pm

```diff
@@ -155,6 +156,83 @@ sub remove_if_needed
     return $?;
 }
 
+
+=pod
+
+=head2 get_disk_uuid
+
+Fetches the (PART)UUID of the device with blkid 
+
+=cut
+
+sub get_disk_uuid
+{
+    my ($self, $type, $device) = @_;
+    
+    my ($cmd_output, $cmd_err);
+    my $cmd = CAF::Process->new([BLKID, $device], log => $this_app, 
+        stdout => \$cmd_output, stderr => \$cmd_err);
+    $cmd->execute();
+    my $re = qr!\s${type}UUID="(\S+)"!m ;
+    if($cmd_output && $cmd_output =~m/$re/m){
+        return $1;
+    }
+}
+
+=pod
+
+=head2 check_in_fstab
+
+Parse the fstab entry for the C<$self> filesystem. When a (PART)UUID is found, 
+this is used instead of the device name or label. If it concerns a strict protected mountpoint or filesystem,
+it will add it, but changes will not be allowed.
+
+=cut
+
+sub check_in_fstab
+{
+    my ($self, $fh, $protected) = @_;
+    my $add = 1;
+    my $ndevice;
+    my $txt = "$fh";
+    my $re = qr!^\s*([^#\s]\S+)\s+$self->{mountpoint}\/?\s+(\S+)\s!m;
+    my $devpath = $self->{block_device}->devpath;
+    while ($txt =~m/$re/mg) {
+        $add = 0;
+        my ($device, $type) = ($1, $3);
+        if ($device =~ /^(PART)?UUID=(\S+)/) {
+            my ($type_uuid, $fstab_uuid) = ($1 || '' , $2);
+            my $qt_uuid = $self->get_disk_uuid($type_uuid, $devpath);
+            if (!$qt_uuid) {
+                $this_app->warn("$type_uuid", "UUID of device $devpath for $self->{mountpoint} in Quattor could not be found");
+                next;
+            } elsif ($qt_uuid ne $fstab_uuid) {
+                $this_app->warn("${type_uuid}UUID of disk in Quattor is different from the ${type_uuid}UUID in the fstab file!");
+            }
+            $ndevice = "${type_uuid}UUID=$qt_uuid";
+        } 
+    }
+    if ($add) {
```

Comment:
please use 1 return at the end of the method instead of curent 6. you also return strings and integers, you shoudl describe this in the pod when this retruns what